### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by luis-novoa**
We finished coding the game logic and we wait for feedback.

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,48 +1,140 @@
...
+    elsif ($board[2] == $board[4])&&($board[4] == $board[6])
+        game_end = 8
+        $winner = $board[2]
+        return true
+    elsif $board.all?(String)
+        game_end = 9
+        return true
+    end
+end
+
```
Comment (ebukaume) [inline bin/main]: This method is way too long.

**Review by ebukaume — CHANGES_REQUESTED** [review]
## Changes Required! :stop_sign: 

Hi @luis-novoa  and @RoodzFernando :wave: 

Good job so far, but there are a few issues we need to address.

- [ ] You are expected to have at least 2 classes. _Usually, three: Game, Board and Player._
pathwright ref
- [ ] All classes are kept in `lib` directory.
pathwright ref
- [ ] The code should respect the single responsibility rule. You `bin/main` file is doing too much.
ref

_See below for more :point_down:_ 

**Kindly submit another code review request after making the changes**

Happy Coding :v: 
**Ebuka Umeokonkwo**
 [Github](https://github.com/ebukaume) | [Linkedin](https://www.linkedin.com/in/ebukaume) | [Slack](https://app.slack.com/team/UFDKXE68L) | [Twitter](https://twitter.com/ebukaume)

File: bin/main [hunk 2]
```diff
@@ -1,48 +1,112 @@
...
+                board_1.board_update($player_2.sign, $player_2.move)
+                break 
+            else
+                puts "Position already taken! Choose again!"
+            end
+        end
+        game_checker = game_finish?(board_1)
+        board_1.board_show
+        break if game_checker
+    end
```
Comment (imhta) [inline bin/main]: **Try to minimize the length of the function. (because it is too large)**

**Review by imhta — CHANGES_REQUESTED** [review]
# Require changes 🛑 

### Hello @luis-novoa @RoodzFernando , 👋   you have done a good job but still there function which is too large. Try to reduce that. Because it is not good practice. reference to that code below.

---
#### Don't forgot to request for another code review 
---
**Pon Muthu Selvam N**
_If you have any queries, feel free to ping me on slack_

**Comment by RoodzFernando** [comment]
We minimized the length of the new_game method.

**Comment by shshamim63** [comment]
## Status :recycle: [!! Required Changes !!] :recycle: 
___
Hi @luis-novoa  and @RoodzFernando :wave,
Nice job :clap:. You have covered all the changes asked by the previous reviewer. That is awesome. But I have noticed some other issues. And I am sure that you will cover them easily :sunglasses: .
___
##### Please follow the instructions below :point_down:  to complete this project.
___
### :spades:  [Mandatory Changes] :spades: 
___
1. You have to move the pointed code from board class. Cause only the main should be responsible for rendering all the output.
https://github.com/RoodzFernando/Tic-Tac-Toe-project/blob/aed994ea5662fce9d3b5e2bc6b07b5fed2744ec2/lib/board.rb#L9-L17
image1
2. You have declared attr_accessor :board. So to access it you don't need that @board anymore.  
https://github.com/RoodzFernando/Tic-Tac-Toe-project/blob/aed994ea5662fce9d3b5e2bc6b07b5fed2744ec2/lib/board.rb#L19-L25
___
### Please fix the above mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/) | [Slack](Shakhawat Hossain)

**Review by Forison — CHANGES_REQUESTED** [review]
### Changes Required🛑
# Hi team👏,
I tried to run your app but kept getting an error🤔,
```require': cannot load such file -- ./lib/board (LoadError)``` is suspect module importation is problematic  
For the sake of your next submission make sure you 
* [ ] The application does not not allow a player to move into an occupied position.
* [ ] The application does not exit due to invalid input - it informs player about it and asks for their move again.
* [ ] The application can determine if move is a winning move
* [ ] The application can tell if there was a draw after players move
* [ ] The application can change active player
### Kindly get it to run on your local machine and test the requirements I have sent and submit for another review

#### Happy coding
Addo Boakye Forison

**Review by Redvanisation — APPROVED** [review]
# Milestone approved ✔️ 

### Hello team 🌟 

### _Nice work on this project, it seems that you have met the requirements so I am approving your project. 👍_ 

### You may merge to master and submit a project completion form. 🚀 

### Good luck! 🍀

**Pull request merged**